src/main/java/io/jenkins/plugins/ml/model/ParsableFile.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2020 Loghi Perinpanayagam.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package io.jenkins.plugins.ml.model;
+
+import hudson.Extension;
+import hudson.Util;
+import hudson.model.AbstractDescribableImpl;
+import hudson.model.Descriptor;
+import hudson.util.FormValidation;
+import hudson.util.ListBoxModel;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.QueryParameter;
+
+import javax.annotation.Nonnull;
+
+public class ParsableFile extends AbstractDescribableImpl<ParsableFile> {
+
+    enum SourceCodeConvertType{
+        // Type of source code to be handled by IPython builder
+        NBFORMAT,
+        ZFORMAT,
+        JSON,
+        PY,
+        NONE
+    }
+
+    private final String filePath;
+    private final boolean deleteFilesAfterBuild;
+    private final String sType;
+
+
+    @DataBoundConstructor
+    public ParsableFile(String filePath, boolean deleteFilesAfterBuild, String sType){
+        this.filePath = filePath;
+        this.deleteFilesAfterBuild = deleteFilesAfterBuild;
+        this.sType = sType;
+    }
+
+    public String getsType() {
+        return sType;
+    }
+
+    public String getFilePath() {
```

Comment:
Probably better to call it `fileName` instead of `filePath`. When I was reading the code in `FileParser` I thought it weird when I saw `FilePath copyFrom = new FilePath(new File(file.getFilePath()));`.